collection/migrate.go

```diff
@@ -42,6 +43,42 @@ func MigrateRepoStoreToLocalCollectionSet(ctx context.Context, s Set, r repo.Rep
 				datasets[i].MetaTitle = ds.Meta.Title
 			}
 		}
+
+		ulog, err := book.UserDatasetBranchesLog(ctx, datasets[i].InitID)
+		if err != nil {
+			log.Errorf("can't get user log for dataset %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		dlog, err := ulog.Log(datasets[i].InitID)
+		if err != nil {
+			log.Errorf("can't get dataset log for dataset %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		if len(dlog.Logs) < 0 {
+			log.Errorf("no branch logs for dataset log %s, %s", datasets[i].InitID, err)
+			continue
+		}
+		blog := dlog.Logs[0]
```

Comment:
This would be much "cleaner" if logbook easily returned the "main" branch log based on an InitID, and then that `BranchLog` had a method `CommitOps` and `RunOps`, that returned a list of ops of just the `CommitModel` and `RunModel` respectively.

However, this is the first time we've needed something like that, and the place we need it is in migration code, whose whole purpose is to pull the information in a way that is not typically expected.

I'm trying to spell out that I don't mind that there isn't a convenient way to get a specific Op from logbook, since the whole point of `collection` is to retain the important information about the dataset, rendering the need to pull it from logbook mostly mute (except in this specific case).

Reply:
agreed. In situations like this, create an unexported function in this file and call it here, making all of this a subroutine that we can consider moving into logbook if it's needed elsewhere